src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -531,35 +531,42 @@ private BlittableJsonReaderObject RecreateCountersIfNeeded(DocumentsOperationCon
             }
         }
 
-        private void DeleteOldRevisions(DocumentsOperationContext context, Table table, Slice prefixSlice, CollectionName collectionName,
+        private bool DeleteOldRevisions(DocumentsOperationContext context, Table table, Slice prefixSlice, CollectionName collectionName,
             RevisionsCollectionConfiguration configuration, long revisionsCount, NonPersistentDocumentFlags nonPersistentFlags, string changeVector, long lastModifiedTicks)
         {
+            var moreRevisionToDelete = false;
             if ((nonPersistentFlags & NonPersistentDocumentFlags.FromSmuggler) == NonPersistentDocumentFlags.FromSmuggler)
-                return;
+                return false;
 
             if (configuration.MinimumRevisionsToKeep.HasValue == false &&
                 configuration.MinimumRevisionAgeToKeep.HasValue == false)
-                return;
+                return false;
 
             long numberOfRevisionsToDelete;
             if (configuration.MinimumRevisionsToKeep.HasValue)
             {
-                numberOfRevisionsToDelete = revisionsCount - configuration.MinimumRevisionsToKeep.Value;
+                numberOfRevisionsToDelete = Math.Min(revisionsCount - configuration.MinimumRevisionsToKeep.Value,
```

Comment:
Can this be negative?

Reply:
yes

Reply:
isn't that a problem?

Reply:
we check if it is negative a few lines later and then return false and don't delete